content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the execution slot | `<In Name="Start">` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+
+### Out
+
+Specifies an execution exit slot
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the execution slot | `<Out Name="Done">` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+
+### SerializedProperty
+
+A C++ member variable that needs to be serializEd
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the execution slot | `<SerializedProperty Name="m_numericPrecision"/>` |
+
+
+### EditProperty
+
+Allows providing __EditContext__ attributes to an existing __Property__ or __SerializedProperty__
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name to replace the field's name, cosmetic only | `<SerializedProperty Name="m_numericPrecision"/>` |
+| __FieldName__ | Required | The name of the __Property__ or __SerializedProperty__ | `FieldName="m_numericPrecision"` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __UIHandler__ | Optional | Allows the user to override the property's default UI Handler in the Node Inspector | `UIHandler="AZ::Edit::UIHandlers::ComboBox"`|
+| __DisplayGroup__ | Optional | Allows slots to be grouped visually | `DisplayGroup="Tests"` |
+
+```xml
+<EditProperty UiHandler="AZ::Edit::UIHandlers::Default" FieldName="m_numericPrecision" Name="Precision" Description="The precision with which to print any numeric values.">
+    <EditAttribute Key="AZ::Edit::Attributes::Min" Value="0"/>
+    <EditAttribute Key="AZ::Edit::Attributes::Max" Value="24"/>
+</EditProperty>
+```
+
+### EditAttribute
+
+__EditProperty__ may also use [EditAttribute](#editattribute)
+
+### DynamicDataSlot
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the data slot | `Name="Format"` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __ConnectionType__ | Required | `ScriptCanvas::ConnectionType::Input` or `ScriptCanvas::ConnectionType::Output` | `ConnectionType="ScriptCanvas::ConnectionType::Input"` |
+| DynamicType | Required | `Value`, `Container`, or `Any` | `DynamicType="ScriptCanvas::DynamicDataType::Container"` | 
```

Comment:
It would be very helpful to have a description on what DynamicDataSlots are. 

Additionally, it would help to have description on the attributes **ConnectionType** and **DynamicType**. For **ConnectionType**, it may be enough for the user to understand what DynamicDataSlots are, in order to figure out whether to choose an input or output Connection type. For DynamicType, when should a user choose value, container, or any?

Reply:
Agree with adding a short description on what Dynamic Data Slots are. And include a link to the topic: `[Dynamic Data Slots](./dynamic-data-slots)`.

Reply:
Agreed, will add it